libraries/stdlib/src/kotlin/collections/Sets.kt

```diff
@@ -93,3 +94,30 @@ internal fun <T> Set<T>.optimizeReadOnlySet() = when (size) {
     1 -> setOf(iterator().next())
     else -> this
 }
+
+/**
+ * Build a new read-only [Set] with the [elements][E] from the [builderAction].
+ *
+ * @sample samples.collections.Builders.Sets.buildSetSample
+ */
+@kotlin.internal.InlineOnly
+@SinceKotlin("1.4")
+@Suppress("NEWER_VERSION_IN_SINCE_KOTLIN", "API_NOT_AVAILABLE" /* See KT-30129 */) // TODO: remove this in 1.4
```

Comment:
You can use `@SinceKotlin("1.3")` + `@ExperimentalStdlibApi`. This way new functions could be published in 1.3.x release